evals/solvers/providers/google/gemini_solver_wav.py

```diff
@@ -0,0 +1,164 @@
+                    str(gen_content_resp.prompt_feedback),
+                    error=gen_content_resp.prompt_feedback,
+                )
+            else:
+                # Get text response
+                solver_result = SolverResult(
+                    gen_content_resp.text,
+                    error=gen_content_resp.prompt_feedback,
+                )
+        except (google.api_core.exceptions.GoogleAPIError,) as e:
+            solver_result = SolverResult(
+                e.message,
+                error=e,
+            )
+        except ValueError as e:
+            # TODO: Why does this error ever occur and how can we handle it better?
+            # (See google/generativeai/types/generation_types.py for the triggers)
+            known_errors = [
+                "The `response.text` quick accessor",
+                "The `response.parts` quick accessor",
+            ]
+            if any(err in str(e) for err in known_errors):
+                solver_result = SolverResult(
+                    str(e),
+                    error=e,
+                )
+            else:
+                raise e
+
+        return solver_result
+
+    @staticmethod
+    def _convert_msgs_to_google_format(msgs: list[Message]) -> list[GoogleMessage]:
+        """
+        Convert messages to Gemini API format and process audio data.
+        """
+        """
+        Example msgs:
+        [Message(role='user', content='You are a helpful assistant.', 
+        tool_calls=None, tool_call_id=None), 
+        Message(role='user', 
+        content=[{'type': 'text', 'text': 'Please translate the text to English. Your response should only include the English translation, without any additional words:\n\n'}, 
+        {'type': 'audio_url', 'audio_url': {'url': 'data:audio/x-wav;base64,Ukl...'}}])
+        """
+        std_msgs = []
+        for msg in msgs:
+            gmsg = GoogleMessage.from_evals_message(msg)
+            assert gmsg.role in {"user", "model"}, f"Invalid role: {gmsg.role}"
+            
+            if std_msgs and gmsg.role == std_msgs[-1].role:
+                # Combine text content
+                if isinstance(gmsg.parts[0], str):
+                    std_msgs[-1].parts = ["\n".join(std_msgs[-1].parts + gmsg.parts)]
+                # Combine text and preserve audio data
+                elif isinstance(gmsg.parts[0], dict):
+                    std_msgs[-1].parts = [{
+                        "type": "text",
+                        "text": std_msgs[-1].parts[0] + "\n" + gmsg.parts[0]["text"]
+                    }, gmsg.parts[1]]
+            else:
+                std_msgs.append(gmsg)
+
+        # Process final audio message
+        last_msg = std_msgs[-1].parts
+        audio_part = last_msg[1]
+        
+        # Convert audio to wav format
+        wav_data = {
+            "mime_type": "audio/wav",
+            "data": _data_url_to_wav(audio_part["audio_url"]["url"])
+        }
```

Comment:
So a multi-turn conversation is not allowed?
Is that a limitation of Gemini?

Reply:
Yup, I'll add a TODO here for handling multi turn convos